### PR TITLE
allow to build shared or static library with position independent code support (fPIC)

### PR DESCRIPTION
**Pull request by m4ce**
I have the need to build both static and shared libraries with fPIC support.

### CODE REVIEW

**Comment by jkhoogland** [comment]
I have the same requirement.
Would be great to have this merged.

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,7 +45,12 @@ set( HEADER_FILES
 	${HEADER_FOLDER}/date/tz_private.h
 )
 
-add_library( tz ${HEADER_FILES} ${SOURCE_FOLDER}/tz.cpp )
+if (BUILD_SHARED_LIBS)
```
Comment (NeroBurner) [inline CMakeLists.txt]: no need for this if-else block, `add_library` recognizes `BUILD_SHARED_LIBS`

> If no type is given explicitly the type is STATIC or SHARED based on whether the current value of the variable BUILD_SHARED_LIBS is ON

https://cmake.org/cmake/help/v3.14/command/add_library.html?highlight=add_library

Comment (m4ce) [reply CMakeLists.txt]: Thanks - amended.

**Comment by HowardHinnant** [comment]
Closing.  Thank you for this good suggestion.  I am in the process of ending cmake support for this library.  I encourage interested parties to host their own cmake files for this library, and I would be happy to link to such support in my README.md.

**Pull request closed without merging**